proxy/deploy.go

```diff
@@ -22,25 +22,35 @@ type FunctionResourceRequest struct {
 	Requests *stack.FunctionResources
 }
 
+// DeployFunction will first check if the function exists, and if it does, or the update bool is set to true, will perform a rolling update
+// otherwise, the function will be deployed as new
 func DeployFunction(fprocess string, gateway string, functionName string, image string,
 	registryAuth string, language string, replace bool, envVars map[string]string,
 	network string, constraints []string, update bool, secrets []string,
 	labels map[string]string, functionResourceRequest1 FunctionResourceRequest) {
 
 	rollingUpdateInfo := fmt.Sprintf("Function %s already exists, attempting rolling-update.", functionName)
-	statusCode, deployOutput := Deploy(fprocess, gateway, functionName, image, registryAuth, language, replace, envVars, network, constraints, update, secrets, labels, functionResourceRequest1)
 
-	if update == true && statusCode == http.StatusNotFound {
-		// Re-run the function with update=false
-		_, deployOutput = Deploy(fprocess, gateway, functionName, image, registryAuth, language, replace, envVars, network, constraints, false, secrets, labels, functionResourceRequest1)
-	} else if statusCode == http.StatusOK {
+	existing := functionExists(gateway, functionName)
+
+	if update {
```

Comment:
When `update=false` and `replace=true` we would want to do a force update, rather than rejecting the command?

Reply:
Good point Eric! Here, I'm only focusing on the `update` flag since update and replace are effectively the same thing. So, if you don't want to update the function (by setting `update=false`), but the function exists, I want to make sure that you really do want to update it... The default value is `true` for the `update` flag, so if you, the user, is explicitly setting the flag to false, we don't want to update/replace that function unexpectedly.

What do you guys think @alexellis, @johnmccabe, and @rgee0?

Reply:
This is about refactoring to address what is essentially a bug where a message is displayed twice in particular scenarios.  The refactor should aim to address the issue without changing the CLI behaviour beyond not outputting the message twice.